Change to flash code

johnw004 opened the pull request:
Checked for master ram board when jumping to OS.

johnw004 committed:
Update flash.asm

Check whether master ram board is present when jumping into reset routine in OS

AtomicRoland commented:
John, you're using &27F for checking the presence of a Master Ram Board. This address is officially unused in the Electron, however, how can we assure that bit 7 of this address is not set by another application? That would make the flash command jump to the wrong address on an Electron without a MRB. Is there a second test for the MRB presence?

johnw004 committed:
Update flash.asm

Changed reset code to work with master ram board regardless of whether any other application has used the master ram flag at &27F.

AtomicRoland commented:
I have implemented the latest changes info my current development version. You should be able to upgrade your ROM to version 0.24 by performing an *UPDATE. 

I did an update from 0.23 to 0.24. The logo was presented correctly on my Elk with Plus-1 and Pres Plus 1 ROM. Didn't test it with one of the other Plus-1 ROMS. Back downgrading to 0.23 also worked so the reset routine in 0.24 is working correctly. After that I upgraded again to 0.24.

Thanks for your contribution.
